### PR TITLE
[Spec] ShuffleChannels spec update

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
There was a restriction in the spec that data input has to be 4D, 
but v0::ShuffleChannels ng op allows input shapes with `rank >= 1D` . 
Such cases are also supported by CPU and some other plugins.

**The main change proposed in this PR is to relax the 4D input restriction and allow ND (1D or bigger) input** 
as alignment with the current version of ShuffleChannels nGraph op, CPU and partially other plugins implementation.

The tests are added with the ShuffleChannels op revision PR: https://github.com/openvinotoolkit/openvino/pull/5764

### Tickets:
 - 44622 (45790)

### CODE REVIEW

File: docs/ops/movement/ShuffleChannels_1.md [hunk 1]
```diff
@@ -8,43 +8,57 @@
...
+
+    dim 0: data_shape[0] * data_shape[1] * ... * data_shape[axis-1] (or 1 if axis == 0)
+    dim 1: group
+    dim 2: data_shape[axis] / group
+    dim 3: data_shape[axis+1] * data_shape[axis+2] * ... * data_shape[data_shape.size()-1] (or 1 if axis points to last dimension)
+
+So the dimensions before and after the dimension pointed by `axis` are flattened and reshaped back to the original shape after channnels shuffling.
+
+
+The operation is equivalent with the following transformation of the input tensor *x* of shape *[N, C, H, W]*:
```
Comment (lazarevevgeny) [inline docs/ops/movement/ShuffleChannels_1.md:23]: Tensor names and shapes should be wrapped with back ticks.

`The operation is equivalent to the following transformation...`

File: docs/ops/movement/ShuffleChannels_1.md [hunk 2]
```diff
@@ -8,43 +8,57 @@
...
-  * **Range of values**: a positive integer
+  * **Range of values**: a positive integer in the range `[1, data_shape[axis]]`
   * **Type**: `int`
   * **Default value**: 1
   * **Required**: *No*
 
 **Inputs**:
 
-*   **1**: 4D input tensor of any supported data type. Required.
+*   **1**: `data` ND (1D or bigger) input tensor of type *T*. **Required.**
```
Comment (lazarevevgeny) [inline docs/ops/movement/ShuffleChannels_1.md:53]: do not use word `ND`. Rewrite like this:
```
`data` input tensor  of type *T* and rank greater or equal to 1.
```

File: docs/ops/movement/ShuffleChannels_1.md [hunk 3]
```diff
@@ -8,43 +8,57 @@
...
 **Short description**: *ShuffleChannels* permutes data in the channel dimension of the input tensor.
 
+**Detailed description**:
+
+Input ND tensor of `data_shape` is always interpreted as 4D tensor with the following shape:
+
+    dim 0: data_shape[0] * data_shape[1] * ... * data_shape[axis-1] (or 1 if axis == 0)
+    dim 1: group
+    dim 2: data_shape[axis] / group
+    dim 3: data_shape[axis+1] * data_shape[axis+2] * ... * data_shape[data_shape.size()-1] (or 1 if axis points to last dimension)
```
Comment (lazarevevgeny) [inline docs/ops/movement/ShuffleChannels_1.md:18]: I think that this text will be rendered ugly. Can we wrap the text with some style modifier for pretty displaying?

File: docs/ops/movement/ShuffleChannels_1.md [hunk 4]
```diff
@@ -8,43 +8,57 @@
 
 **Short description**: *ShuffleChannels* permutes data in the channel dimension of the input tensor.
 
+**Detailed description**:
+
+Input ND tensor of `data_shape` is always interpreted as 4D tensor with the following shape:
```
Comment (lazarevevgeny) [inline docs/ops/movement/ShuffleChannels_1.md:13]: Do not use word `ND`.

File: docs/ops/movement/ShuffleChannels_1.md [hunk 5]
```diff
@@ -8,43 +8,57 @@
...
+
+    dim 0: data_shape[0] * data_shape[1] * ... * data_shape[axis-1] (or 1 if axis == 0)
+    dim 1: group
+    dim 2: data_shape[axis] / group
+    dim 3: data_shape[axis+1] * data_shape[axis+2] * ... * data_shape[data_shape.size()-1] (or 1 if axis points to last dimension)
+
+So the dimensions before and after the dimension pointed by `axis` are flattened and reshaped back to the original shape after channnels shuffling.
```
Comment (lazarevevgeny) [inline docs/ops/movement/ShuffleChannels_1.md:20]: Trailing and leading to `axis` dimensions are flattened...

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: docs/ops/movement/ShuffleChannels_1.md (hunk 1 above)
Comment (lazarevevgeny) [inline docs/ops/movement/ShuffleChannels_1.md:23]: And `axis = 1`

File: docs/ops/movement/ShuffleChannels_1.md [hunk 6]
```diff
@@ -8,43 +8,57 @@
...
+
 **Attributes**:
 
 * *axis*
 
   * **Description**: *axis* specifies the index of a channel dimension.
-  * **Range of values**: an integer number in the range [-4, 3]
+  * **Range of values**: an integer number in the range `[-rank(data_shape), rank(data_shape) - 1]`
   * **Type**: `int`
   * **Default value**: 1
```
Comment (ggalieroc) [inline docs/ops/movement/ShuffleChannels_1.md:40]: Should values also be enclosed between backticks?

File: docs/ops/movement/ShuffleChannels_1.md (hunk 5 above)
Comment (ggalieroc) [inline docs/ops/movement/ShuffleChannels_1.md:20]: nitpick: _channnels_ -> channels

**Review by jdanieck — APPROVED** [review]
No more comments than already provided by others.

File: docs/ops/movement/ShuffleChannels_1.md (hunk 6 above)
Comment (mitruska) [reply docs/ops/movement/ShuffleChannels_1.md:40]: Do you mean  `**Default value**: 1` or ``[-rank(data_shape), rank(data_shape) - 1]`` ?
I would say the current version is ok. Could you explain your suggestion please?

Comment (ggalieroc) [reply docs/ops/movement/ShuffleChannels_1.md:43]: yes I referred to Default value, as for bool types I have seen values true and false surrounded by ``

File: docs/ops/movement/ShuffleChannels_1.md (hunk 3 above)
Comment (mitruska) [reply docs/ops/movement/ShuffleChannels_1.md]: The tab on the begining of each line work as style modifier and make it a separate pseudo-code like block.
I think it's not so ugly now;)


image

File: docs/ops/movement/ShuffleChannels_1.md (hunk 6 above)
Comment (mitruska) [reply docs/ops/movement/ShuffleChannels_1.md:43]: I can see in other ops spec (e.g. [LogSoftmax_5.md](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/activation/LogSoftmax_5.md) or [Proposal_4.md](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/detection/Proposal_4.md)) that it's not a rule for numeric values but ofc I can change it, if we decide to be consistent with that.

File: docs/ops/movement/ShuffleChannels_1.md (hunk 1 above)
Comment (mitruska) [reply docs/ops/movement/ShuffleChannels_1.md]: Done

File: docs/ops/movement/ShuffleChannels_1.md (hunk 2 above)
Comment (mitruska) [reply docs/ops/movement/ShuffleChannels_1.md]: Done

File: docs/ops/movement/ShuffleChannels_1.md (hunk 4 above)
Comment (mitruska) [reply docs/ops/movement/ShuffleChannels_1.md]: Done

File: docs/ops/movement/ShuffleChannels_1.md (hunk 1 above)
Comment (mitruska) [reply docs/ops/movement/ShuffleChannels_1.md]: Done

File: docs/ops/movement/ShuffleChannels_1.md (hunk 5 above)
Comment (mitruska) [reply docs/ops/movement/ShuffleChannels_1.md]: Done

Comment (mitruska) [reply docs/ops/movement/ShuffleChannels_1.md]: Done

**Comment by jdanieck** [comment]
@tomdol maybe you'll have time to review & merge to offload @lazarevevgeny ? Unreviewed PR's started to pile up because of the release. CC: @andrei-kochin

**Review by ggalieroc — APPROVED** [review]

**Review by tomdol — APPROVED** [review]

**Comment by tomdol** [comment]
Cannot merge because github says that Evgeny requested changes. @lazarevevgeny please r-review when ready.

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**